ballerina/utils.bal

```diff
@@ -0,0 +1,108 @@
+// Copyright (c) 2024, WSO2 LLC. (https://www.wso2.com).
+//
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+isolated function prettifyJson(json value, string indentation, int level, boolean isMapField = false) returns string {
+    if value == () {
+        return " null";
+    } else if value is map<json> {
+        return prettifyJsonMap(value, indentation, level, isMapField);
+    } else if value is json[] {
+        return prettifyJsonArray(value, indentation, level, isMapField);
+    } else {
+        return prettifyJsonField(value, indentation, level, isMapField);
+    }
+}
+
+isolated function prettifyJsonMap(map<json> value, string indentation, int level, boolean isMapField) returns string {
+    string initialIndentation = getInitialIndentation(indentation, level, isMapField);
+    string result = string `${initialIndentation}{`;
+    boolean isEmptyMap = value.keys().length() == 0;
+    if !isEmptyMap {
+        result += "\n";
```

Comment:
Adding `\n` to the string may cause failures in windows, isn't it?

Reply:
Windows tests are passing in the PR build. is there a recommended way to handle this?